crates/chia-client/src/request_map.rs

```diff
@@ -0,0 +1,65 @@
+use std::{collections::HashMap, sync::Arc};
+
+use chia_protocol::Message;
+use tokio::sync::{oneshot, Mutex, OwnedSemaphorePermit, Semaphore};
+
+#[derive(Debug)]
+pub struct Request {
+    sender: oneshot::Sender<Message>,
+    _permit: OwnedSemaphorePermit,
+}
+
+impl Request {
+    pub fn send(self, message: Message) {
+        self.sender.send(message).ok();
+    }
+}
+
+#[derive(Debug)]
+pub struct RequestMap {
+    items: Mutex<HashMap<u16, Request>>,
+    semaphore: Arc<Semaphore>,
+}
+
+impl RequestMap {
+    pub fn new() -> Self {
+        Self {
+            items: Mutex::new(HashMap::new()),
+            semaphore: Arc::new(Semaphore::new(u16::MAX as usize)),
```

Comment:
can you add a comment explaining why you initialize the semaphore to `u16::MAX`?
it seems like some arbitrarily picked upper limit

Reply:
In the Chia protocol, message ids are 16 bit unsigned integers